### DISCUSSION TITLE
How do I generate a prisma client in my current project when the schema file is in another project?

### DISCUSSION THREAD

**Discussion by jume-dev** [prisma/prisma#17764] [Q&A] [answered]
Hi there,
I struggle to find a way to make the question from the title work.
This is a mono-repo setting. There is project A containing migrations and the schema.prisma file at the default location, so pretty default project.
There is project B where I want to generate the client to actually use it in the code. I added
```
  "prisma": {
    "schema": "../projectA/prisma/schema.prisma"
  }
```
to the package.json.
While this works the output is now in `projectA/node_modules/.prisma` instead of `projectB/node_modules/.prisma`, which is surprising because I would assume, it uses the current work dir for determining the output path instead of the schema file location.

Any idea on how to fix that? Did I miss some documentation or discussion on this?
Thanks

**Comment by nurul3101** [comment]
Hello @jume-dev 👋 

Can you try configuring [custom output path](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/generating-prisma-client#the-location-of-prisma-client) for generating the PrismaClient?
You can then import the generated prisma client from the custom path that you have configured.

**Reply by jume-dev** [reply]
mhm, but what if projectC comes along and also wants to have that client? The solution feels a bit unflexible. I'd prefer if I maybe could add the output path as a parameter to the generate cli command

**Reply by nurul3101** [reply]
Do you mean something similar to providing a schema path while invoking the `npx prisma generate` command?
[Reference](https://www.prisma.io/docs/reference/api-reference/command-reference#generate-prisma-client-using-a-non-default-schemaprisma-path)

**Reply by jume-dev** [reply]
yes, exactly. That would be great

**Comment by arosemena** [comment, marked as answer]
You could just use a simple copy command `npx prisma generate && cp -r node_modules/.prisma otherProjectPath/node_modules/.prisma`

If you add it as a package json scripts you could call it with something like `npm run generate` or `yarn generate`

**Reply by jume-dev** [reply]
yeah, that work, but it makes the whole build process in our build pipeline a little more complex. But for now it's manageable.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
